File: 2015/08/puzzle1.js
var fs = require('fs');

fs.readFile('day8.txt', 'utf-8', function (err, inp) {
 if (err) throw err;

 var chars = 0, mem = 0;
 inp.trim().split(/\n/g).forEach(function (line) {
  var d = line.match(/^\"(.*)\"$/,line)
  if (!d) {
   return;
  }
  chars += line.length;
  var re = /(\\\\|\\"|\\x([0-9a-f]{2})|(.+?))/gi;
  var s = d[1], m;
  while (m = re.exec(s)) {
   mem += m[1][0] == '\\' ? 1 : m[1].length;
  }
 });
 console.log(chars - mem);
});

File: 2015/08/puzzle1.py
import re

chars = 0
mem = 0

def repl(p):
 return {
  r'\"': lambda x: '"',
  r'\\': lambda x: '\\',
  r'\x': lambda x: x.decode('hex')
 }[filter(lambda x: x is not None, [p.group(2), p.group(4), p.group(5)])[0]](p.group(3))

for i in open('day8.txt'):
 line = i.strip()
 chars += len(line)
 m = re.match('^"(.*)"$', line)
 if m != None:
  s = re.sub(r'((\\x)([0-9A-F]{2})|(\\")|(\\\\))', repl, m.group(1), flags=re.I)
  mem += len(s)
print chars - mem
